src/pilot/assign_role.py

```diff
@@ -846,7 +846,30 @@ def assign_role(ip_mac_service_tag, node_uuid, role_index,
     if role_index.index:
         role = "node:{}-{}".format(flavor, role_index.index)
 
-    value = "{},{},boot_mode:uefi,boot_option:local".format(role, node.properties['capabilities'])
+    # capabilities will be available if in-band introspection or
+    # out-of-band inspection is performed before
+    # assign_role.py is executed.
+    # This conditional handling is needed to run this script before
+    # out-of-band inspection or in-band introspection has been performed.
+    if 'capabilities' in node.properties:
+        current_capabilities = node.properties.get('capabilities')
+        try:
+            cap_dict = dict(x.split(':', 1)
+                            for x in current_capabilities.split(','))
+        except ValueError:
+        # ValueError can occur if capabilities values not properly
+        # defined like capabilities='boot_mode:uefi,boot_option'.
+            raise ValueError(
+                    ("Invalid capabilities string '%s'.") % current_capabilities)
```

Comment:
1. fix indentation to be 4 spaces from previous line
2. remove () around `("Invalid capabilities string '%s'.")` - not necessary after removing `_`